avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
+                               sha.sha(fingerprint).hexdigest()[:4])
```

Comment:
The hash is quite important as we'll be using it (along with the name) as a base for the per-variant-data-dir and the content of such `data-dir` could depend not just on the names, but also on the values.

Do you really believe it'd be frequent to have clashes in complex examples? (we are not looking for security, only for convenient way to distinguish between value changes so it's fine to be able to produce clashes on purpose, but it should not be frequent accidentally).